docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,28 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when you want to expose a setting
+for the user to use his own tool-specific config. In case the user specifies
+such a config file, we can avoid generating one again with ``generate_config``
+to reduce I/O load.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+        @staticmethod
+        def generate_config(filename, file, user_config, setting_a: bool):
```

Comment:
I think `user_config` should contain default value here, too.

Reply:
Also, `setting_a` should have a default value. Look https://github.com/coala/coala/pull/4899#discussion_r168275071

Reply:
Okay. Let me do that.